Doc/library/asyncio-pools.rst

```diff
@@ -0,0 +1,95 @@
+.. currentmodule:: asyncio
+
+.. versionadded:: 3.9
+
+.. _asyncio-pools:
+
+=====
+Pools
+=====
+
+**Source code:** :source:`Lib/asyncio/pools.py`
+
+-----------------------------------------------
+
+.. note::
+    This section of the documentation and all of its members have been
+    added *provisionally* to asyncio's API. For more details, see
+    :term:`provisional api`.
+
+Asyncio pools are high-level, asynchronous context managers that can be used
+to concurrently run blocking functions and methods.
+
+There are many potential use cases, but a particularly useful one is for
+combining libraries without asyncio support with existing asyncio programs.
+Normally, calling a non-async function within the event loop would will result
+in blocking the event loop until the function returns. However, by using a
+pool to run the function, it can be executed in a separate worker (such as a
+thread or process) without blocking the event loop.
+
+.. class:: ThreadPool(concurrency=None)
+
+   An asynchronous threadpool that provides methods to concurrently
+   run IO-bound functions, without blocking the event loop.
+
+   *concurrency* is an optional argument that limits the number of
+   threads to utilize in the threadpool. With the default value of
+   ``None``, the amount of threads used will scale based on the
+   number of processors.
+
+   .. coroutinemethod:: run(/, func, *args, **kwargs)
```

Comment:
Is there a reason to deviate from the `Executor` API (which calls this method `submit`)? Do you fear people might use one for another?

Reply:
I don't personally have especially strong views on using `run()` vs `submit()`, @1st1 initially suggested the usage of `run()` being the primary means of executing functions within the thread pool (in the BPO issue). But, it does seem to be more aligned with the existing `run_in_executor()` that this was based on.

Perhaps Yury could elaborate on why he initially chose `run()` over `submit()` as the method name?

Reply:
For additional reference, another pool class, `multiprocessing.pool.Pool`, uses `apply` as a name.

Reply:
```suggestion
   .. coroutinemethod:: run(func, /, *args, **kwargs)
```